staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -401,8 +423,15 @@ func ServerGroupsAndResources(d DiscoveryInterface) ([]*metav1.APIGroup, []*meta
 	for i := range sgs.Groups {
 		resultGroups = append(resultGroups, &sgs.Groups[i])
 	}
+	// resources is non-nil if aggregated discovery succeeded.
 	if resources != nil {
-		return resultGroups, resources, nil
+		// Any stale Group/Versions returned by aggregated discovery
+		// must be surfaced to the caller as failed Group/Versions.
+		var ferr error
+		if len(failedGVs) > 0 {
+			ferr = &ErrGroupDiscoveryFailed{Groups: failedGVs}
```

Comment:
@Jefftree Example: `ErrGroupDiscoveryFailed` encapsulates the stale GV's when it is returned here.

Reply:
Ah got it, it's still encapsulated. thank you!